src/hypothesis/strategies.py

```diff
@@ -735,6 +736,92 @@ def binary(
 
 @cacheable
 @defines_strategy
+def domains(
+    alphabet=None, unsafe=None, min_subdomains=None, max_subdomains=None,
```

Comment:
I chose alphabet since I'm using it to generate multiple strings of text, each with different length, and the text strategy imposes a min/max limit already. Is there another way to do this? I also think that parameter is needed to be able to constrain the emails to just ASCII or no special characters, etc.